modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesRecoveryTask.java

```diff
@@ -23,129 +23,197 @@
+			List<AndesExchange> exchangeList = amqpConstructStore.getExchanges();
+			List<AndesExchange> duplicatedExchanges = new ArrayList<>(exchangesStored);
+
+			exchangesStored.removeAll(exchangeList);
+			for (AndesExchange exchange : exchangesStored) {
+				for (ExchangeListener listener : exchangeListeners) {
+					log.warn("Recovering node. Adding exchange " + exchange.toString());
+					listener.handleClusterExchangesChanged(exchange,
+														   ExchangeListener.ExchangeChange.Added);
+				}
+			}
+
+			exchangeList.removeAll(duplicatedExchanges);
+			for (AndesExchange exchange : exchangeList) {
+				for (ExchangeListener listener : exchangeListeners) {
+					log.warn("Recovering node. Removing exchange " + exchange.toString());
+					listener.handleClusterExchangesChanged(exchange,
+														   ExchangeListener.ExchangeChange.Deleted);
+				}
+			}
+		} else {
+			log.info("Failed to recover exchanges from database due to non-operational context store.");
+		}
+	}
+
+	private void reloadQueuesFromDB() throws AndesException {
+		if(isContextStoreOperational.get()) {
+			List<AndesQueue> queuesStored = andesContextStore.getAllQueuesStored();
+			List<AndesQueue> queueList = amqpConstructStore.getQueues();
+			List<AndesQueue> duplicatedQueues = new ArrayList<>(queuesStored);
+
+			queuesStored.removeAll(queueList);
+			for (AndesQueue queue : queuesStored) {
+				for (QueueListener listener : queueListeners) {
+					log.warn("Recovering node. Adding queue " + queue.toString());
+					listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.ADDED);
+				}
+			}
+
+			queueList.removeAll(duplicatedQueues);
+			for (AndesQueue queue : queueList) {
+				for (QueueListener listener : queueListeners) {
+					log.warn("Recovering node. Removing queue " + queue.toString());
+					listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.DELETED);
+				}
+			}
+		} else {
+			log.info("Failed to recover queues from database due to non-operational context store.");
+		}
+	}
+
+	private void reloadBindingsFromDB() throws AndesException {
+		if(isContextStoreOperational.get()) {
+			List<AndesExchange> exchanges = andesContextStore.getAllExchangesStored();
+			for (AndesExchange exchange : exchanges) {
+				List<AndesBinding> bindingsStored =
+						andesContextStore.getBindingsStoredForExchange(exchange.exchangeName);
+				List<AndesBinding> bindingsForExchange =
+						amqpConstructStore.getBindingsForExchange(exchange.exchangeName);
+				List<AndesBinding> duplicatedBindings = new ArrayList<>(bindingsStored);
+				bindingsStored.removeAll(bindingsForExchange);
+				for (AndesBinding binding : bindingsStored) {
+					for (BindingListener listener : bindingListeners) {
+						log.warn("Recovering node. Adding binding " + binding.toString());
+						listener.handleClusterBindingsChanged(binding,
+															  BindingListener.BindingEvent.ADDED);
```

Comment:
Broke the line to avoid exceeding standard code line length. Checked the code in IDE and there is no problem in the indentation.